pkg/api/v1alpha1/cluster_webhook.go

```diff
@@ -107,6 +112,36 @@ func (r *Cluster) ValidateUpdate(old runtime.Object) error {
 	return nil
 }
 
+func validateBundlesRefCluster(new, old *Cluster) field.ErrorList {
+	var allErrs field.ErrorList
+	bundlesRefPath := field.NewPath("spec").Child("BundlesRef")
+
+	if old.Spec.BundlesRef != nil && new.Spec.BundlesRef == nil {
+		allErrs = append(
+			allErrs,
+			field.Invalid(bundlesRefPath, new.Spec.BundlesRef, fmt.Sprintf("field cannot be removed after setting. Previous value %v", old.Spec.BundlesRef)))
+	}
+
+	if old.Spec.BundlesRef != nil && new.Spec.BundlesRef != nil && standardBundleRef.MatchString(old.Spec.BundlesRef.Name) && standardBundleRef.MatchString(new.Spec.BundlesRef.Name) {
```

Comment:
is it necessary to run the regex twice per bundle? Can't we consolidate it into one?
When running `FindStringSubmatch`, if the matches lenght is not 2, then we know it doesn't match. We can either return -1 from the `getBundleIdFromBundleRefName` (common in the go standard library) or we could even return `(bool, int)`

Reply:
Refactored to only run regex once per bundle